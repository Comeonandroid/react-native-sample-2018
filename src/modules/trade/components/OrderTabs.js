import React, { PureComponent } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import { View } from '../../../elements'

import OrderTab from './OrderTab'

class OrderTabs extends PureComponent {
  render() {
    const { tabs } = this.props
    return (
      <TabsContainer>
        {tabs.map((tab, i) => (
          <OrderTab 
            key={tab.value}
            active={this.props.active === tab.value}
            last={this.props.tabs.length - 1 === i}
            {...tab} 
            onTabPress={this.props.onTabPress}
          />
        ))}
      </TabsContainer>
    )
  }
}

OrderTabs.propTypes = {
  tabs: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string
  })),
  onTabPress: PropTypes.func,
  active: PropTypes.string
}

const TabsContainer = styled(View)`
  display: flex;
  flex-direction: row;
`

export default OrderTabs
