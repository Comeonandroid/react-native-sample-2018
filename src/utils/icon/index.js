import { createIconSetFromIcoMoon } from 'react-native-vector-icons'
import styled from 'styled-components'

import icoMoonConfig from './selection.json'
const CustomIcon = createIconSetFromIcoMoon(icoMoonConfig)

export default styled(CustomIcon) `
  color: ${ ({ theme, color }) => color || theme.colors.border };
`

// IMPORTANT
// If you want to add/remove icons please read this first
// https://www.reactnative.guide/12-svg-icons-using-react-native-vector-icons/12.1-creating-custom-iconset.html