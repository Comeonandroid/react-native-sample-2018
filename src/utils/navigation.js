import React from 'react'

import { t } from '../locales/i18n'

import HeaderMenuIcon from '../modules/navigation/header/components/HeaderMenuIcon'

export const getNavigationHeader = ({ navigation, screenProps }) => (
  {
    headerMode: 'float',
    title: t('route.' + navigation.state.routeName),
    headerTintColor: screenProps.theme.colors.headerContrastColor,
    headerStyle: {
      backgroundColor: screenProps.theme.colors.background,
      borderBottomColor: 'transparent',
    },
    headerRight: <HeaderMenuIcon navigation={navigation} />,
    // TODO: header title for all the pages
    headerBackTitle: null
  }
)

export const getHomeHeader = ({ navigation, screenProps }) => (
  {
    headerTintColor: screenProps.theme.colors.headerContrastColor,
    headerStyle: {
      backgroundColor: screenProps.theme.colors.background,
      borderBottomColor: 'transparent',
    },
    headerRight: <HeaderMenuIcon navigation={navigation} />
  }
)
