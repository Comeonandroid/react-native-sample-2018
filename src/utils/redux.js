import {
  createReactNavigationReduxMiddleware,
  createReduxBoundAddListener,
} from 'react-navigation-redux-helpers'

const reduxRouterMiddleware = createReactNavigationReduxMiddleware(
  "root",
  state => state.navigation,
)
const addListener = createReduxBoundAddListener('root')

export {
  reduxRouterMiddleware,
  addListener,
}
