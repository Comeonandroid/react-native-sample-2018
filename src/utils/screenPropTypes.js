import {
  PropTypes
} from 'prop-types'

export default {
  navigation: PropTypes.object.isRequired
}