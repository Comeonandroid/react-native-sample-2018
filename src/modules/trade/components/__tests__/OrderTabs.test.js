import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { itBehavesLike } from '../../../../specs/shared_examples/index'
import { wrapWithTheme } from '../../../../specs/utils'

import OrderTabs from '../OrderTabs'

const setup = propOverrides => {
  const props = Object.assign({
    tabs: [
      { label: 'label 1', value: 'value 1' },
      { label: 'label 2', value: 'value 2' }
    ],
    onTabPress: jest.fn(),
    active: 'value 1'
  }, propOverrides)

  const component = <OrderTabs {...props} />
  const wrapper = shallow(component)
  const tree = renderer.create(wrapWithTheme(component))

  return {
    props,
    wrapper,
    tree,
    component
  }
}

describe('<OrderTabs />', () => {
  const { props, component, wrapper, tree } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  expect(wrapper.children().length).toEqual(props.tabs.length)
  expect(tree.root.findByProps({ value: props.active }).props.active).toBeTruthy()

  describe('active: value 2', () => {
    const { component, tree } = setup({ active: 'value 2' })
    itBehavesLike('aThemedComponent', { subject: component })

    expect(tree.root.findByProps({ value: 'value 2' }).props.active).toBeTruthy()
  })
})
