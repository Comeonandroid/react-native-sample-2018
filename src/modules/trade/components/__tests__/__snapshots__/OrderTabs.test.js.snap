// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderTabs /> active: value 2 dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#000000",
        "borderRightWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={false}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcfcfc",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "backgroundColor": "#464242",
              "color": "#c8c8c8",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 1
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#000000",
        "borderRightWidth": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcfcfc",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "color": "#faa632",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 2
    </Text>
  </View>
</View>
`;

exports[`<OrderTabs /> active: value 2 light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#fefaec",
        "borderRightWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={false}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "backgroundColor": "#eae8e0",
              "color": "#000000",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 1
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#fefaec",
        "borderRightWidth": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "color": "#faa632",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 2
    </Text>
  </View>
</View>
`;

exports[`<OrderTabs /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#000000",
        "borderRightWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcfcfc",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "color": "#faa632",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 1
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#000000",
        "borderRightWidth": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={false}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcfcfc",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "backgroundColor": "#464242",
              "color": "#c8c8c8",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 2
    </Text>
  </View>
</View>
`;

exports[`<OrderTabs /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ],
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#fefaec",
        "borderRightWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "color": "#faa632",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 1
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#fefaec",
        "borderRightWidth": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      active={false}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "backgroundColor": "#eae8e0",
              "color": "#000000",
              "paddingBottom": 4,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 4,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    >
      label 2
    </Text>
  </View>
</View>
`;
